"Spack / macos-latest" CI failing

The macOS variant of the Spack build in CI seems to be failing on several PRs.

See, for example, https://github.com/UIUC-PPL/charm/runs/5237462709?check_suite_focus=true.

It seems like the issue is that `+setcpuaffinity` is not having the desired effect of allowing for oversubscription when running for this build, as this error snippet from the linked log shows: 

```
2022-02-17T20:37:15.9801540Z ../../bin/testrun   +p4 ./check 4 ++local +setcpuaffinity
2022-02-17T20:37:16.0187180Z 
2022-02-17T20:37:16.0371560Z Running as 4 OS processes: ./check 4 +setcpuaffinity
2022-02-17T20:37:16.0472190Z charmrun> mpirun -np 4 ./check 4 +setcpuaffinity
2022-02-17T20:37:16.1589470Z --------------------------------------------------------------------------
2022-02-17T20:37:16.1690330Z There are not enough slots available in the system to satisfy the 4
2022-02-17T20:37:16.1790190Z slots that were requested by the application:
2022-02-17T20:37:16.1890250Z 
2022-02-17T20:37:16.1990970Z   ./check
2022-02-17T20:37:16.2091080Z 
2022-02-17T20:37:16.2192320Z Either request fewer slots for your application, or make more slots
2022-02-17T20:37:16.2292590Z available for use.
2022-02-17T20:37:16.2392660Z 
2022-02-17T20:37:16.2493420Z A "slot" is the Open MPI term for an allocatable unit where we can
2022-02-17T20:37:16.2594160Z launch a process.  The number of slots available are defined by the
2022-02-17T20:37:16.2694380Z environment in which Open MPI processes are run:
2022-02-17T20:37:16.2794380Z 
2022-02-17T20:37:16.2894990Z   1. Hostfile, via "slots=N" clauses (N defaults to number of
2022-02-17T20:37:16.2995490Z      processor cores if not provided)
2022-02-17T20:37:16.3096170Z   2. The --host command line parameter, via a ":N" suffix on the
2022-02-17T20:37:16.3196170Z      hostname (N defaults to 1 if not provided)
2022-02-17T20:37:16.3296920Z   3. Resource manager (e.g., SLURM, PBS/Torque, LSF, etc.)
2022-02-17T20:37:16.3397740Z   4. If none of a hostfile, the --host command line parameter, or an
2022-02-17T20:37:16.3497590Z      RM is present, Open MPI defaults to the number of processor cores
2022-02-17T20:37:16.3598110Z 
2022-02-17T20:37:16.3698970Z In all the above cases, if you want Open MPI to default to the number
2022-02-17T20:37:16.3788350Z of hardware threads instead of the number of processor cores, use the
2022-02-17T20:37:16.3864130Z --use-hwthread-cpus option.
2022-02-17T20:37:16.3965130Z 
2022-02-17T20:37:16.4008110Z Alternatively, you can use the --oversubscribe option to ignore the
2022-02-17T20:37:16.4099640Z number of available slots when deciding the number of processes to
2022-02-17T20:37:16.4138280Z launch.
2022-02-17T20:37:16.4202290Z --------------------------------------------------------------------------
2022-02-17T20:37:16.4244750Z 
2022-02-17T20:37:16.4283170Z real	0m0.254s
2022-02-17T20:37:16.4349850Z user	0m0.019s
2022-02-17T20:37:16.4400600Z sys	0m0.020s
2022-02-17T20:37:16.4447090Z make[1]: *** [test] Error 1
2022-02-17T20:37:16.4517920Z make: *** [test-util] Error 2
2022-02-17T20:37:16.4863340Z ==> Error: ProcessError: Command exited with status 2:
2022-02-17T20:37:16.4965080Z     'make' '-C' '/var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/runner/spack-stage/spack-stage-charmpp-main-tqv6udhsgyw345itsik55euriloo7j4m/spack-src/tests' 'test' 'TESTOPTS=++local +setcpuaffinity'
2022-02-17T20:37:19.1892410Z 
2022-02-17T20:37:19.1993230Z 2 errors found in build log:
2022-02-17T20:37:19.2093550Z      6105    launch.
2022-02-17T20:37:19.2194950Z      6106    ------------------------------------------------------------------
2022-02-17T20:37:19.2295100Z              --------
2022-02-17T20:37:19.2351550Z      6107    
2022-02-17T20:37:19.2407330Z      6108    real	0m0.254s
2022-02-17T20:37:19.2478560Z      6109    user	0m0.019s
2022-02-17T20:37:19.2579730Z      6110    sys	0m0.020s
2022-02-17T20:37:19.2680090Z   >> 6111    make[1]: *** [test] Error 1
2022-02-17T20:37:19.2735440Z   >> 6112    make: *** [test-util] Error 2
2022-02-17T20:37:19.2816970Z 
2022-02-17T20:37:19.2917880Z See build log for details:
2022-02-17T20:37:19.3020180Z   /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/runner/spack-stage/spack-stage-charmpp-main-tqv6udhsgyw345itsik55euriloo7j4m/spack-build-out.txt
2022-02-17T20:37:19.3020640Z 
2022-02-17T20:37:26.2656330Z ##[error]Process completed with exit code 1.
```

--- a/.github/workflows/spack.yml
+++ b/.github/workflows/spack.yml
@@ -41,9 +41,9 @@ jobs:
 
     - name: Build Charm++
       run: |
-        export OMPI_MCA_rmaps_base_oversubscribe=1
         cd ../spack
         source share/spack/setup-env.sh
         spack compiler find
         spack compiler list
-        spack install -v --test root charmpp@main
+        spack spec charmpp@main
+        spack install --test root charmpp@main

--- a/.github/workflows/spack.yml
+++ b/.github/workflows/spack.yml
@@ -41,6 +41,7 @@ jobs:
 
     - name: Build Charm++
       run: |
+        export OMPI_MCA_rmaps_base_oversubscribe=1
         cd ../spack
         source share/spack/setup-env.sh
         spack compiler find

--- a/.github/workflows/spack.yml
+++ b/.github/workflows/spack.yml
@@ -45,5 +45,5 @@ jobs:
         source share/spack/setup-env.sh
         spack compiler find
         spack compiler list
-        spack spec charmpp@main
-        spack install --test root charmpp@main
+        spack spec charmpp@main backend=multicore
+        spack install --test root charmpp@main backend=multicore

